src/Microsoft.Health.Dicom.Core/Features/CustomTag/CustomTagEntry.cs

```diff
@@ -3,10 +3,12 @@
 // Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
 // -------------------------------------------------------------------------------------------------
 
+using System;
+
 namespace Microsoft.Health.Dicom.Core.Features.CustomTag
 {
     /// <summary>
-    /// Represent each custom tag entry from customer input.
+    /// External representation of a custom tag entry.
     /// </summary>
     public class CustomTagEntry
```

Comment:
>CustomTagEntry [](start = 17, length = 14)

If the properties are "settable", I don't think constructors are really necessary. That said, can we make this entity immutable? I would probably recommend 1 of 2 ways:
1. Given the size of this entry, it may be reasonable to use a `readonly` `struct` here
2. Otherwise, we could use the new C# 9 record type, which is a fancy wrapper around `class`, to auto-generate the equality, hash-code, etc. There is a new tutorial page [here]([https://docs.microsoft.com/en-us/dotnet/csharp/tutorials/exploration/records#:~:text=C#%209%20introduces%20records,%20a%20new%20reference%20type,classes%20in%20that%20record%20types%20use%20value-based%20equality](https://docs.microsoft.com/en-us/dotnet/csharp/tutorials/exploration/records#:~:text=C#%209%20introduces%20records,%20a%20new%20reference%20type,classes%20in%20that%20record%20types%20use%20value-based%20equality).)

Reply:
I think there is only one scenario where we require a settable property (Status has to be updateable).  The rest of them I've made get only.

---
In reply to: [564843380](https://github.com/microsoft/dicom-server/pull/501#discussion_r564843380) [](ancestors = 564843380)